### ISSUE TITLE
Can't upload a file

### ISSUE THREAD

**Issue by AndrewPiterov** [JannikLassahn/graphql-dotnet-upload#28]
Hi, thanks for library,

I try to upload a single file, but get the error below 

![2023-01-25_17-37-34](https://user-images.githubusercontent.com/10525569/214607173-99a5998d-f3ae-4e2e-ab54-85b4d5df3a1b.png)

I use .NET 7 and following packages

```
<PackageReference Include="GraphQL" Version="7.2.2" />
<PackageReference Include="GraphQL.Server.Authorization.AspNetCore" Version="7.2.0" />
<PackageReference Include="GraphQL.Server.Transports.AspNetCore" Version="7.2.0" />
<PackageReference Include="GraphQL.Server.Ui.Altair" Version="7.2.0" />
<PackageReference Include="GraphQL.SystemTextJson" Version="7.2.2" />
<PackageReference Include="GraphQL.Upload.AspNetCore" Version="3.0.1" />
```

registered as 

```csharp
builder.Services.AddGraphQLUpload();
builder.Services.AddGraphQL();

app.UseGraphQLUpload<ISchema>();
app.UseGraphQL<ISchema>();
app.UseGraphQLAltair();
```

Thanks for any help.

**Comment by AndrewPiterov** [comment]
After delete `Content-Type=application/json` from Headers I've started to  get error below 
![2023-01-25_19-11-47](https://user-images.githubusercontent.com/10525569/214633174-68fb63d0-1021-4243-a8f0-06552df065e0.png)

**Comment by AndrewPiterov** [comment]
After I commented `graphQlBuilder.AddAuthorizationRule();` file uploading started to work, but Authorization is broken now.

```csharp
builder.Services.AddGraphQL(
    graphQlBuilder =>
    {
        // require that the user be authenticated
        graphQlBuilder.AddAuthorizationRule(); // <<<<<<<<<<<< COMMENTED THIS LINE
        graphQlBuilder.AddErrorInfoProvider(opt => opt.ExposeExceptionDetails = true);
        // Adds all graph types in the current assembly with a singleton lifetime.
        graphQlBuilder.AddGraphTypes();
```
The question now is "How to enable authorization and not broke file uploading?"

**Comment by JannikLassahn** [comment]
Hi,
I just released version 3.0.2 which fixes the user not being set in the execution options.
You should now be able to use authorization as usual.

For context: With a call to `AddAuthorizationRule`  the server package hooks up a validation rule that will perform the authorization checks. As far as I can tell, validation rules are registered as singletons within GraphQL, so this library gets to use this rule automatically as well.

**Comment by AndrewPiterov** [comment]
🙏 Thank you, @JannikLassahn !

Now it's working as expected 🚀

I only leave one more comment that GraphQL stuff should be registered after Authorization stuff, e.g.:

```c#
...
app.UseAuthentication();
app.UseAuthorization();
....
app.UseGraphQLUpload<ISchema>();
app.UseGraphQL<ISchema>();
```
